### PR TITLE
Javadoc

### PR DESCRIPTION
**Pull request by schaffe**

### CODE REVIEW

File: src/com/backendless/Messaging.java [hunk 1]
```diff
@@ -332,6 +332,16 @@ public void getRegistrations( AsyncCallback<DeviceRegistration> responder )
     Invoker.invokeAsync( DEVICE_REGISTRATION_MANAGER_SERVER_ALIAS, "getDeviceRegistrationByDeviceId", new Object[] { Backendless.getApplicationId(), Backendless.getVersion(), DEVICE_ID }, responder );
   }
 
+  /**
+   * Publishes message to "Default" channel. The message is not a push notification, it does not have any headers and
+   * does not go into any subtopics.
+   *
+   * @param   message   object to publish. The object can be of any data type - a primitive value, String, Date, a
+   *                    user-defined complex type, a collection or an array of these types.
+   * @return  ${MessageStatus} - a data structure which contains ID of the published message and the status of the
```
Comment (scadgek) [inline src/com/backendless/Messaging.java]: I think this should be {@link MessageStatus}

File: src/com/backendless/Messaging.java [hunk 2]
```diff
@@ -340,17 +350,65 @@ public MessageStatus publish( Object message ) throws BackendlessException
...
                                 PublishOptions publishOptions ) throws BackendlessException
   {
     return publish( channelName, message, publishOptions, new DeliveryOptions() );
   }
 
+  /**
+   * Publishes message to specified channel.The message may be configured as a push notification. It may have headers
+   * and/or subtopic defined in the publishOptions argument.
+   *
+   * @param   channelName name of the channel to publish the message to. If the channel does not exist, Backendless
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: name of A channel..

File: src/com/backendless/Messaging.java (hunk 1 above)
Comment (markpiller) [inline src/com/backendless/Messaging.java]: Right, ${MessageStatus} does not look right

File: src/com/backendless/Messaging.java [hunk 3]
```diff
@@ -340,17 +350,65 @@ public MessageStatus publish( Object message ) throws BackendlessException
...
+   * defined in the publishOptions argument.
+   *
+   * @param   channelName name of the channel to publish the message to. If the channel does not exist, Backendless
+   *                      automatically creates it.
+   * @param   message     object to publish. The object can be of any data type - a primitive value, String, Date, a
+   *                      user-defined complex type, a collection or an array of these types.
+   * @param   publishOptions
+   *                      an instance of ${PublishOptions}. When provided may contain publisher ID (an arbitrary,
+   *                      application-specific string value identifying the publisher), subtopic value and/or a
+   *                      collection of headers.
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: I would also mention what those headers are used for

File: src/com/backendless/Messaging.java [hunk 4]
```diff
@@ -340,17 +350,65 @@ public MessageStatus publish( Object message ) throws BackendlessException
...
+   * @param   deliveryOptions
+   *                      an instance of DeliveryOptions. When provides may specify options for message delivery such
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: when provideD...

File: src/com/backendless/Messaging.java [hunk 5]
```diff
@@ -378,11 +436,43 @@ private String getCheckedChannelName( String channelName )
     return channelName;
   }
 
+  /**
+   * Publishes message to "Default" channel. The message is not a push notification, it does not have any headers and
```
Comment (markpiller) [inline src/com/backendless/Messaging.java]: Please check if it is "Default" or "default"

File: src/com/backendless/Messaging.java (hunk 1 above)
Comment (schaffe) [inline src/com/backendless/Messaging.java]: thanks

**Pull request merged**